app/jobs/invoices/finalize_all_job.rb

```diff
@@ -6,7 +6,13 @@ class FinalizeAllJob < ApplicationJob
 
     def perform(organization:, invoice_ids:)
       result = Invoices::FinalizeBatchService.new(organization:).call(invoice_ids)
-      result.raise_if_error!
+      result.raise_if_error! unless tax_error?(result.error)
```

Comment:
Error returned from finalize service is validation error so you have to check it a bit different: `result.error.messages.dig(:tax_error)`. Also, we need it in `refresh_draft_job.rb` and `finalize_job.rb`

Reply:
fixed :)